backend/FwLite/FwLiteWeb/Routes/ProjectRoutes.cs

```diff
@@ -53,11 +53,12 @@ public static IEndpointConventionBuilder MapProjectRoutes(this WebApplication ap
             async (IOptions<AuthConfig> options,
                 CombinedProjectsService combinedProjectsService,
                 string code,
-                string serverAuthority
+                string serverAuthority,
+                [FromQuery] UserProjectRole? role
             ) =>
             {
                 var server = options.Value.GetServerByAuthority(serverAuthority);
-                await combinedProjectsService.DownloadProject(code, server);
+                await combinedProjectsService.DownloadProjectByCode(code, server, role);
```

Comment:
this API currently always returns ok, we should at the very least return a 500 if the result is not success.

Reply:
Done in commit cef859bb4.